src/Tizen.NUI.Components/Controls/Notification.cs

```diff
@@ -0,0 +1,394 @@
+/*
+ * Copyright(c) 2020 Samsung Electronics Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+using System;
+using System.ComponentModel;
+using Tizen.NUI.BaseComponents;
+
+namespace Tizen.NUI.Components
+{
+    /// <summary>
+    /// Notification helps to raise a notification window with a content View.
+    /// </summary>
+    /// <privilege>http://tizen.org/privilege/window.priority.set</privilege>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class Notification
+    {
+        private Window notificationWindow;
+
+        private Timer timer;
+
+        private Timer extraTimer;
+
+        private NotificationLevel level = NotificationLevel.Base;
+
+        private Rectangle positionSize;
+
+        private bool dismissOnTouch = false;
+
+        private OnPostDelegate onPost;
+
+        private OnDismissDelegate onDismiss;
+
+        private NotificationState state = NotificationState.Ready;
+
+        /// <summary>
+        /// Create a notification with a content View.
+        /// </summary>
+        /// <exception cref="ArgumentException">Thrown when a given contentView is invalid.</exception>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public Notification(View contentView)
+        {
+            if (contentView == null)
+            {
+                throw (new ArgumentException("Input contentView should not be null."));
+            }
+
+            ContentView = contentView;
+        }
+
+        /// <summary>
+        /// Delegate to be called on Notification post.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public delegate void OnPostDelegate(Notification notification);
+
+        /// <summary>
+        /// Delegate to be called on Notification dismiss.
+        /// The return value means delay dismissal in ms (0 means dismiss directly).
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public delegate uint OnDismissDelegate(Notification notification);
+
+        private enum NotificationState
+        {
+            Ready,
+            Post,
+            DelayingOnDismiss,
```

Comment:
What is the meaning of DelayingOnDismiss?

Reply:
Please visit [here](https://github.com/dalihub/nui-demo/tree/master/wearable-samples/Notification/FullsizeNotification#define-ondismiss-behavior) to see onDismiss return value usage.

User can delay dismiss by defining onDismiss delegate.